igvc_perception/include/pointcloud_filter/fast_segment_filter/fast_segment_filter.h

```diff
@@ -0,0 +1,101 @@
+#ifndef GROUNDSEGMENTER_H
+#define GROUNDSEGMENTER_H
+
+#include <math.h>
+#include <pcl/point_cloud.h>
+#include <pcl_ros/point_cloud.h>
+#include <pointcloud_filter/fast_segment_filter/fast_segment_filter_config.h>
+#include <pointcloud_filter/filter.h>
+#include <velodyne_pointcloud/point_types.h>
+#include <visualization_msgs/Marker.h>
+#include <visualization_msgs/MarkerArray.h>
+
+#include <Eigen/Dense>
+
+typedef Eigen::Matrix<double, Eigen::Dynamic, 3> MatrixXd;
+typedef Eigen::Matrix<double, Eigen::Dynamic, 1> VectorXd;
+
+namespace pointcloud_filter
+{
+struct Prototype
+{
+  double distance_;
+  velodyne_pointcloud::PointXYZIR point_;
+  double blue_;
+  double red_;
+  double green_;
+
+  bool operator<(const Prototype &p) const
+  {
+    return distance_ < p.distance_;
+  }
+};
+
+struct Line
+{
+  Eigen::Vector3d params_;
+  Eigen::Vector3d intercept_;
+  Prototype start_point_;
+  Prototype end_point_;
+  double error_t_;
+  std::vector<Prototype> model_points_;
+  bool isGround_;
+  double blue_;
+  double red_;
+  double green_;
+
+  Line()
+  {
+  }
+  Line(double threshold)
+  {
+    error_t_ = threshold;
+  }
+  double distFromPoint(const Prototype p);
+  bool fitPoints(const Prototype new_point);
+};
+
+struct Segment
+{
+  std::vector<velodyne_pointcloud::PointXYZIR> raw_points_;
+  std::vector<Prototype> prototype_points_;
+  std::vector<Line> lines_;
+};
+
+class FastSegmentFilter
+{
+public:
+  FastSegmentFilter(const ros::NodeHandle &nh);
+
+  std::map<int, Segment> segments_;
```

Comment:
Correct me if I'm wrong, but it doesn't seem like you're using the sorted property of `std::map`. If that's the case, then `std::unordered_map` is a better fit here as `std::map` isn't actually a hashmap but a rb tree.